src/transaction_simulator.rs

```diff
@@ -470,8 +617,8 @@ fn get_simulated_transaction_result(transaction: &SimulatedTransaction) -> Trans
                 reason: tx.revert_reason.clone(),
             },
         },
-        TransactionTrace::DeployAccount(_) => TransactionResult::DeployAccount,
-        TransactionTrace::L1Handler(_) => TransactionResult::L1Handler,
-        TransactionTrace::Declare(_) => TransactionResult::Declare,
+        TransactionTrace::DeployAccount(_) => TransactionResult::Success,
+        TransactionTrace::L1Handler(_) => TransactionResult::Success,
+        TransactionTrace::Declare(_) => TransactionResult::Success,
```

Comment:
Not sure what Success would look like here for these transactions. Perhaps, checking in `state_diff.deployed_contracts` for a new address/class hash?

Revert reason only exists for Invoke TXs (https://github.com/NethermindEth/juno/blob/main/rpc/trace.go#L47)